app/components/clock/Clock.jsx

```diff
@@ -11,19 +11,40 @@ class Clock extends React.Component {
     };
 
     state = {
-        dateNow: this._getCurrentDate()
+        hours:      this._getCurrentHours(),
+        minutes:    this._getCurrentMinutes(),
+        seconds:    this._getCurrentSeconds()
     };
 
     _updateDate = () => {
         this.setState({
-            dateNow: this._getCurrentDate()
+            hours:      this._getCurrentHours(),
+            minutes:    this._getCurrentMinutes(),
+            seconds:    this._getCurrentSeconds()
         });
     };
 
-    _getCurrentDate() {
+    _getCurrentHours() {
         const date = new Date();
+        const hours_24 = date.getHours();
 
-        return date.toLocaleTimeString(this.props.hourFormat === '12' ? 'en-US' : 'en-GB');
+        let hours_12 = hours_24 % 12;
+
+        hours_12 = hours_12 ? hours_12 : 12;
+
+        return(this.props.hourFormat === '12' ? hours_12 : hours_24);
+    }
+
+    _getCurrentMinutes() {
+        const date = new Date();
+
+        return (date.getMinutes() < 10 ? `0${date.getMinutes()}` : date.getMinutes());
```

Comment:
If you see that some method is called 2 or more time - it is better to introduce *variable* for such case. (in your case `date.getMinutes()` called thrice). This way you will save some processor time and make code more readable.
As well as previous `return` point, *parenthesis* here are redundant.